composition-js/src/merging/merge.ts

```diff
@@ -1081,6 +1081,11 @@ class Merger {
     return this.metadata(sourceIdx).isFieldFullyExternal(field);
   }
 
+  private typeIsFullyExternal(object: ObjectType | InterfaceType): boolean {
+    return object.hasAppliedDirective('external')
+      || object.fields().every(f => f.hasAppliedDirective('external'));
+  }
```

Comment:
A few things here:
- The phrase "fully external" has a special meaning in federation, so we shouldn't use it unless we want that meaning (it means a field that is either marked `@external` or its parent is, but is not in a `@provides` field set and is not in a type extension `@key`). I'd suggest renaming it e.g. to `areAllFieldsExternal()`.
- Directly checking for an `@external` directive application won't deal with renames appropriately. To deal with that, we'll want to use `FederationMetadata.isFieldExternal()` similar to `Merger.isExternal()` above (note that function checks for parent type `@external`s appropriately).

What I'd suggest is something like

```suggestion
  private areAllFieldsExternal(sourceIdx: number, type: ObjectType | InterfaceType): boolean {
    return type.fields().every(f => this.metadata(sourceIdx).isFieldExternal(f));
  }
```

Then at the callsite, you'll want to provide `sourceIdx`.